tests/testscript_long_migration_with_load.py

```diff
@@ -0,0 +1,137 @@
+        computeVM.execute(
+            'virsh list --name | while read domain; do [[ -n "$domain" ]] && virsh destroy "$domain"; done'
+        )
+        computeVM.execute(
+            'virsh list --all --name | while read domain; do [[ -n "$domain" ]] && virsh undefine "$domain"; done'
+        )
+
+        # After undefining and destroying all domains, there should not be any .xml files left
+        # Any files left here, indicate that we do not clean up properly
+        controllerVM.fail("find /run/libvirt/ch -name *.xml | grep .")
+        controllerVM.fail("find /var/lib/libvirt/ch -name *.xml | grep .")
+        computeVM.fail("find /run/libvirt/ch -name *.xml | grep .")
+        computeVM.fail("find /var/lib/libvirt/ch -name *.xml | grep .")
+
+        # Destroy any remaining huge page allocations.
+        controllerVM.succeed("echo 0 > /proc/sys/vm/nr_hugepages")
+        computeVM.succeed("echo 0 > /proc/sys/vm/nr_hugepages")
+
+        # Remove any remaining vm logs.
+        controllerVM.succeed("rm -f /tmp/*.log")
+        computeVM.succeed("rm -f /tmp/*.log")
+
+    def test_live_migration_long_running_with_load(self):
+        """
+        This test performs 500 back-and-forth live migrations in a row.
+        During live-migration, the VM is under memory load with a working set
+        of roughly 1.6GiB.
+        """
+
+        controllerVM.succeed("virsh -c ch:///session define /etc/domain-chv.xml")
+        controllerVM.succeed("virsh -c ch:///session start testvm")
+
+        assert wait_for_ssh(controllerVM)
+
+        status, _ = ssh(controllerVM, "screen -dmS stress stress -m 4 --vm-bytes 400M")
+        assert status == 0
+
+        run_loops = 500
+        for i in range(run_loops):
+            print(f"Run {i+1}/{run_loops}")
+
+            controllerVM.succeed(
+                "virsh -c ch:///session migrate --domain testvm --desturi ch+tcp://computeVM/session --persistent --live --p2p"
+            )
+            assert wait_for_ssh(computeVM)
+
+            computeVM.succeed(
+                "virsh -c ch:///session migrate --domain testvm --desturi ch+tcp://controllerVM/session --persistent --live --p2p"
+            )
+            assert wait_for_ssh(controllerVM)
+
+def suite():
+    suite = unittest.TestSuite()
+    suite.addTest(LibvirtTests("test_live_migration_long_running_with_load"))
+    return suite
+
+def wait_for_ssh(machine, user="root", password="root", ip="192.168.1.2"):
+    retries = 100
+    for i in range(retries):
+        print(f"Wait for ssh {i}/{retries}")
+        status, _ = ssh(machine, "echo hello", user, password, ip="192.168.1.2")
+        if status == 0:
+            return True
+        time.sleep(1)
+    return False
+
+
+def ssh(machine, cmd, user="root", password="root", ip="192.168.1.2"):
+    status, out = machine.execute(
+        f"sshpass -p {password} ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no {user}@{ip} {cmd}"
+    )
+    return status, out
```

Comment:
You can put the `ssh` and `wait_for_ssh` method into a separate file and import them here and in the other test script to reduce a little duplication.

Reply:
Ah that won't work the way the test script is loaded in the test at the end. nevermind for now